pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -697,14 +697,28 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 
 		if corev1.PodSucceeded != pod.Status.Phase {
 			if corev1.PodFailed == pod.Status.Phase {
-				c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPodFailed, "Populator failed: %s", pod.Status.Message)
-				// Delete failed pods so we can try again
-				if !plan.RestartLimitExceeded(pod) {
-					err = c.kubeClient.CoreV1().Pods(populatorNamespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+				restarts, ok := pvc.Annotations[AnnPopulatorReCreations]
+				if !ok {
+					pvc.Annotations[AnnPopulatorReCreations] = "1"
+					err = c.restartFailedPopulator(ctx, pvc, populatorNamespace, pod.Name)
 					if err != nil {
 						return err
 					}
+					return nil
 				}
+				restartsInteger, err := strconv.Atoi(restarts)
+				if err != nil {
+					return err
+				}
+				if restartsInteger < 3 {
+					pvc.Annotations[AnnPopulatorReCreations] = strconv.Itoa(restartsInteger + 1)
+					err = c.restartFailedPopulator(ctx, pvc, populatorNamespace, pod.Name)
+					if err != nil {
+						return err
+					}
+					return nil
+				}
+				c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPodFailed, "Populator failed: Please check the logs in populator pod, %s/%s", populatorNamespace, pod.Name)
```

Comment:
```suggestion
				c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPodFailed, "Populator failed: Please check the logs of the populator pod, %s/%s", populatorNamespace, pod.Name)
```
we can also say that few (3) attempts failed